pkg/kubelet/volumemanager/volume_manager.go

```diff
@@ -332,7 +332,7 @@ func (vm *volumeManager) GetVolumesInUse() []v1.UniqueVolumeName {
 	}
 
 	for _, volume := range allAttachedVolumes {
-		if volume.PluginIsAttachable {
+		if volume.PluginIsAttachable || volume.DeviceMountState != operationexecutor.DeviceNotMounted {
```

Comment:
I wonder if this problematic from another perspective. For example - a pod is scheduled to the node and while it is not mounted yet but mount operation has started (so it is in actual state of the world) but while mount is pending the pod is deleted and hence removed from DSOW. I think this will cause such volumes to be not reported in-use which is problematic I think.

Reply:
We also look at asw.allAttachedVolumes. As long as the map is populated correctly for node stage / node mount operations [MarkDeviceAsMounted](https://github.com/kubernetes/kubernetes/blob/master/pkg/volume/util/operationexecutor/operation_generator.go#L619)  , we would still report the volumes-in-use. Am I missing something here?

Reply:
There could be a window that volume is neither in actual state nor desired state (operation is pending). I kind of remember I added some logic to make sure the volume should be already in actual state before we can remove it from desired state.. Let me check it.

Reply:
actually this should not be a problem for two reasons. a. We are still checking for attachable volume plugins (for which the volumeInUse check is important) b. We have protection in-place to prevent mounting until volume has been reported as in-use.

Still we should clearly document this here and perhaps we need some tests for this.

Reply:
One more thing (that I am too lazy to verify myself), this change *does not* causes other volume types to be reported as in-use right? Such as we should never report secrets/emptydirs in `volumeInUse` variable.

This change will cause local volumes to be reported as in-use though. I think, that in itself is not a problem since nobody is looking for local volume's `node.Status.VolumeInUse` field. cc @msau42

Reply:
> this change _does not_ causes other volume types to be reported as in-use right? Such as we should never report secrets/emptydirs

I will try it out to check the behavior for secrets and emptyDir

Reply:
I verified that - we are fine. I was being paranoid. But we do report local volumes now. I think reporting local volumes in node's status is mostly harmless.

Reply:
I verified that emptyDir and secret are not reported in volumesInUse